src/TopoMojo.Hypervisor/Proxmox/ProxmoxVnetsClient.cs

```diff
@@ -0,0 +1,188 @@
+using Microsoft.Extensions.Logging;
+using TopoMojo.Hypervisor.Proxmox.Models;
+
+namespace TopoMojo.Hypervisor.Proxmox
+{
+    public interface IProxmoxVnetsClient
+    {
+        Task<IEnumerable<PveVnet>> CreateVnets(IEnumerable<CreatePveVnet> createVnets);
+        Task<IEnumerable<PveVnet>> DeleteVnets(IEnumerable<string> names);
+        Task<IEnumerable<PveVnet>> DeleteVnetsByTerm(string term);
+        Task<IEnumerable<PveVnet>> GetVnets();
+        Task ReloadVnets();
+    }
+
+    public class ProxmoxVnetsClient : IProxmoxVnetsClient
+    {
+        private readonly ILogger<ProxmoxVnetsClient> _logger;
+        private readonly PveClient _pveClient;
+        private readonly Random _random;
+
+        public ProxmoxVnetsClient
+        (
+            HypervisorServiceConfiguration hypervisorOptions,
+            ILogger<ProxmoxVnetsClient> logger,
+            Random random
+        )
+        {
+            _logger = logger;
+            _pveClient = new PveClient(hypervisorOptions.Host, 443)
+            {
+                ApiToken = hypervisorOptions.Password
+            };
+            _random = random;
+        }
+
+        public async Task<IEnumerable<PveVnet>> CreateVnets(IEnumerable<CreatePveVnet> createVnets)
+        {
+            var existingNets = await this.GetVnets();
+            var deployedNets = new List<PveVnet>();
+
+            foreach (var createVnet in createVnets)
+            {
+                if (existingNets.Any(n => n.Alias == createVnet.Alias))
+                {
+                    _logger.LogDebug($"Skipped creating vnet {createVnet} - it already exists.");
+                    continue;
+                }
+
+                var newVnetTag = default(int?);
+
+                do
+                {
+                    newVnetTag = _random.Next(100, 100000);
+                }
+                while (existingNets.Any(n => n.Tag == newVnetTag));
+
+                var vnetId = this.GetRandomVnetId();
+
+                // check for existence of alias = vnetname--gamespaceid
+                var createTask = _pveClient.Cluster.Sdn.Vnets.Create
+                (
+                    vnet: vnetId,
+                    tag: newVnetTag,
+                    zone: createVnet.Zone,
+                    alias: createVnet.Alias
+                ).Result;
+
+                if (createTask.IsSuccessStatusCode)
+                {
+
+                    deployedNets.Add
+                    (
+                        new PveVnet
+                        {
+                            Alias = createVnet.Alias,
+                            Tag = newVnetTag.GetValueOrDefault(),
+                            Type = string.Empty,
+                            Vnet = vnetId,
+                            Zone = createVnet.Zone
+                        }
+                    );
+                }
+            }
+
+            if (deployedNets.Any())
+            {
+                await this.ReloadVnets();
```

Comment:
I don't think we want to ReloadVnets here as it results in a duplicate reload with the one already at the end of a batch